include/exadg/operators/operator_base.cpp

```diff
@@ -521,20 +530,51 @@ OperatorBase<dim, Number, n_components>::add_block_diagonal_matrices(BlockMatrix
 template<int dim, typename Number, int n_components>
 void
 OperatorBase<dim, Number, n_components>::apply_block_diagonal_matrix_based(
```

Comment:
Shouldn't this function, and the other one below for the inverse block matrices, be implemented as a free function? So we need to discuss: Why is `OperatorBase` executing this function? With the changes of this PR, isn't it unsatisfactory that this is still done by `OperatorBase`?